src/jobs/config_sync_job.ts

```diff
@@ -377,6 +489,133 @@ export class ConfigSyncJob extends SyncJob {
+      ? (this._connection.secondService.name === 'Jira' || this._connection.secondService.name === 'Redmine')
+      : false
+  }
+
+  private async _createTicket2TicketIssueMappings(firstServiceObjects: ServiceObject[], secondServiceObjects: ServiceObject[]): Promise<Mapping[]> {
+    const firstServiceIssuesWithCustField = firstServiceObjects.filter((o: ServiceObject) => {
+      return o.syncCustomFieldValue
+    })
+    const secondServiceIssuesWithCustField = secondServiceObjects.filter((o: ServiceObject) => {
+      return o.syncCustomFieldValue
+    })
+    const newMappings: Mapping[] = new Array()
+    const firstSecond = await this._createTicket2TicketMapping(firstServiceObjects, secondServiceIssuesWithCustField, true)
+    const secondFirst = await this._createTicket2TicketMapping(secondServiceObjects, firstServiceIssuesWithCustField, false)
+    newMappings.push(...firstSecond)
+    newMappings.push(...secondFirst)
+    return newMappings
+  }
+
+  private async _createTicket2TicketMapping(firstServiceObjects: ServiceObject[], secondServiceObjectsWithCustField: ServiceObject[], first: boolean): Promise<Mapping[]> {
+    const newMappings: Mapping[] = new Array()
+    for (const secondObject of secondServiceObjectsWithCustField) {
+      const idOfFirst = await getIdOfAnotherServiceIdFromLink(first ? this._connection.firstService : this._connection.secondService, secondObject.syncCustomFieldValue)
+      if (idOfFirst) {//'second objects has link to the first => first is primary
+        const firstObject = firstServiceObjects.find((o: ServiceObject) => {
+          return o.id == idOfFirst
+        })
+        if (firstObject) {
+          //check if projects corespond with those that should be paired
+          const areInTheProjectPair = this._connection.projectMappings.filter((p: ProjectMapping) => {
+            (p.idFirstService === firstObject.projectId && p.idSecondService === secondObject.projectId)
+              || (p.idFirstService === secondObject.projectId && p.idSecondService === firstObject.projectId)
+          })
+          if (areInTheProjectPair) {
+            const mapping = new Mapping()
+            mapping.primaryObjectId = firstObject.id
+            mapping.primaryObjectType = firstObject.type
+            mapping.name = firstObject.name
+            mapping.mappingsObjects.push(
+              new MappingsObject(
+                firstObject.id,
+                firstObject.name,
+                first ? this._connection.firstService.name : this._connection.secondService.name,
+                firstObject.type)
+            )
+            const secondMappingsObject =
+              new MappingsObject(
+                secondObject.id,
+                secondObject.name,
+                first ? this._connection.secondService.name : this._connection.firstService.name,
+                secondObject.type)
```

Comment:
Proč tohle není uděláno stejně jako první objekt?

Reply:
asi mi to zbylo takto z testování, sjednoceno